src/star-utils.cpp

```diff
@@ -15,24 +18,155 @@ bool CatalogStarMagnitudeCompare(const CatalogStar &a, const CatalogStar &b) {
+    // We technically don't even need the verification set unless we plan
+    // to calculate probability of mismatch - (which we probably should)
+
+    // Definitely keep the first star
+    keepForPatterns[0] = true;
+    keepForVerifying[0] = true;
+
+    for (int i = 1; i < numEntries; i++) {
+        // Spatial vector representing new star
+        Vec3 vec = catalog[i].spatial;
+
+        bool angsPatternsOK = true;
+        int numPattStarsInFov = 0;
+
+        // We should test each new star's angular distance to all stars
+        // we've already selected to be kept for pattern construction
+        // Stop early if:
+        // a) double star: angle < min separation allowed
+        // b) Number of stars in region maxFov/2 >= pattStarsPerFOV
+        for (int j = 0; j < i; j++) {
+            if (keepForPatterns[j]) {
+                float dotProd = vec * catalog[j].spatial;
+                if (dotProd >= std::cos(DegToRad(starMinSep))) {
+                    angsPatternsOK = false;
+                    break;
+                }
+                // If angle between new star i and old star j is less than maxFov/2, OK
+                if (dotProd > std::cos(maxFOV / 2)) {
+                    numPattStarsInFov++;
+                    if (numPattStarsInFov >= pattStarsPerFOV) {
+                        angsPatternsOK = false;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (angsPatternsOK) {
+            keepForPatterns[i] = true;
+            keepForVerifying[i] = true;
+            keepForPattCount++;
+            continue;
+        }
+
+        bool angsVerifyingOK = true;
+        int numVerStarsInFov = 0;
+
+        // Same thing here, we should test each new star's angular distance to all
+        // stars we've already selected to be kept for verification
+        // std::vector<float> angsVerifying;
+        for (int j = 0; j < i; j++) {
+            if (keepForVerifying[j]) {
+                float dotProd = vec * catalog[j].spatial;
+                if (dotProd >= std::cos(DegToRad(starMinSep))) {
+                    angsVerifyingOK = false;
+                    break;
+                }
+                if (dotProd > std::cos(maxFOV / 2)) {
+                    numVerStarsInFov++;
+                    if (numVerStarsInFov >= verificationStarsPerFOV) {
+                        angsVerifyingOK = false;
+                        break;
+                    }
+                }
+            }
+        }
+
+        if (angsVerifyingOK) {
+            keepForVerifying[i] = true;
+        }
+    }
+
+    std::vector<short> finalCatIndices;
+    std::vector<short> pattStars;
+
+    // finalCat is the final version of the star table
+    for (int i = 0; i < (int)keepForVerifying.size(); i++) {
+        if (keepForVerifying[i]) {
+            finalCatIndices.push_back(i);
+        }
+    }
+
+    // Pretty clever way of finding which stars in the FINAL star table
```

Comment:
It's unclear what this loop is doing and honestly it seems wrong, for example if star 0 and 2 are in keepForVerifying, and star 2 is in keepForPatterns, then star 1 (which is in neither) will be added to pattStars?

If this does turn out to be a bug, then it's an indication that the whole routine should probably be unit tested...